tests/testthat/test-lengths_linter.R

```diff
@@ -1,32 +1,44 @@
 test_that("lengths_linter skips allowed usages", {
+  linter <- lengths_linter()
+
+  expect_lint("length(x)", NULL, linter)
+  expect_lint("function(x) length(x) + 1L", NULL, linter)
+  expect_lint("vapply(x, fun, integer(length(y)))", NULL, linter)
+
   # TODO(#1570): also throw a lint here, and for map(x, length)
-  expect_lint("lapply(x, length)", NULL, lengths_linter())
+  expect_lint("lapply(x, length)", NULL, linter)
 })
 
 test_that("lengths_linter blocks simple disallowed base usages", {
-  expect_lint(
-    "sapply(x, length)",
-    rex::rex("Use lengths() to find the length of each element in a list."),
-    lengths_linter()
-  )
-
-  expect_lint(
-    "vapply(x, length, integer(1L))",
-    rex::rex("Use lengths() to find the length of each element in a list."),
-    lengths_linter()
-  )
+  linter <- lengths_linter()
+  lint_msg <- rex::rex("Use lengths() to find the length of each element in a list.")
+
+  expect_lint("sapply(x, length)", lint_msg, linter)
+  expect_lint("sapply(x, FUN = length)", lint_msg, linter)
+  expect_lint("sapply(FUN = length, x)", lint_msg, linter)
+
+  expect_lint("vapply(x, length, integer(1L))", lint_msg, linter)
```

Comment:
similar suggested test: `sapply(x, sqrt, simplify = length(x))` should also trivially pass, but good to keep as a regression test

Reply:
Thanks. Done!